comp/core/agenttelemetry/impl/sender.go

```diff
@@ -161,22 +166,59 @@ func newSenderClientImpl(agentCfg config.Component) client {
 	return httputils.NewResetClient(httpClientResetInterval, httpClientFactory(agentCfg, httpClientTimeout))
 }
 
+// buils url from a config endpoint.
+func buildURL(endpoint logconfig.Endpoint) string {
+	var address string
+	if endpoint.Port != 0 {
+		address = fmt.Sprintf("%v:%v", endpoint.Host, endpoint.Port)
+	} else {
+		address = endpoint.Host
+	}
+	url := url.URL{
+		Scheme: "https",
+		Host:   address,
+		Path:   telemetryPath,
+	}
+
+	return url.String()
+}
+
+func getEndpoints(cfgComp config.Component) (*logconfig.Endpoints, error) {
+	// borrowed and styled after EP Forwarder newHTTPPassthroughPipeline().
+	// Will be eliminated in the future after switching to EP Forwarder.
+	configKeys := logconfig.NewLogsConfigKeys(telemetryConfigPrefix, cfgComp)
+	endpoints, err := logconfig.BuildHTTPEndpointsWithConfig(cfgComp, configKeys,
+		telemetryHostnameEndpointPrefix, telemetryIntakeTrackType, logconfig.DefaultIntakeProtocol, logconfig.DefaultIntakeOrigin)
+	if err != nil {
+		return nil, err
+	}
+
+	// If dual shipping endpoint overrides main endpoint
```

Comment:
In which scenarios is this needed? (similar question: what is the use case for `agent_telemetry.use_only_additional_endpoint`?)

AFAIK no other sender/forwarder in the Agent supports such an option, because there are no use cases for them where the user would want the dual shipping endpoint to override the main endpoint (if a user needs to override the main endpoint without dual-shipping, they can simply use `agent_telemetry.dd_url`, correct?)

Reply:
Good question. You correct in your assumption. I am planning to send agent telemetry only to site where host belongs too. Do it otherwise IMHO would increase the cost but more importantly, I afraid, will skew statistics. If I understand `agent_telemetry.dd_url` correctly it will be sent to with the main/top `api_key`  (there are no api_key overwrite like in additional_endpoint) which in US5 will not solve the problem.

Reply:
> If I understand agent_telemetry.dd_url correctly it will be sent to with the main/top api_key (there are no api_key overwrite like in additional_endpoint) which in US5 will not solve the problem.